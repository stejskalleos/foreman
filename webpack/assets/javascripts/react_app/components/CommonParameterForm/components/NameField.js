import React from 'react';
import PropTypes from 'prop-types';

import { translate as __ } from '../../../common/I18n';
import FormField from '../../common/forms/FormField';

const NameField = ({ value, onChange, error }) => (
  <FormField label={__('Name')} required error={error}>
    <input
      type="text"
      id="common_parameter_name"
      name="common_parameter[name]"
      value={value}
      onChange={e => onChange(e.target.value)}
      className="form-control"
      required
    />
  </FormField>
);

NameField.propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  error: PropTypes.string,
};

NameField.defaultProps = {
  value: '',
  error: '',
};

export default NameField;
