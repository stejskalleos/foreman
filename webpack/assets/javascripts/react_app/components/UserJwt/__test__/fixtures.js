import { noop } from '../../../common/helpers';
import { STATUS } from '../../../constants';

export const formProps = {
  handleSubmit: noop,
};

const token = 'token23';

export const fieldOk = {
  apiStatus: STATUS.RESOLVED,
  token,
};

export const fieldError = {
  apiStatus: STATUS.ERROR,
  token,
};

export const fieldPending = {
  apiStatus: STATUS.PENDING,
  token,
};
