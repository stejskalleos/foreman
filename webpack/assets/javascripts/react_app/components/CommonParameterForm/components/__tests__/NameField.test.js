import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import { nameFieldFixtures } from './fixtures';
import NameField from '../NameField';

describe('NameField', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(NameField, nameFieldFixtures));
});
