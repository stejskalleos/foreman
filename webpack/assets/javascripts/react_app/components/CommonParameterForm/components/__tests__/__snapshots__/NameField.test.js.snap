// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameField rendering default 1`] = `
<FormField
  className=""
  disabled={false}
  error=""
  helpInline={null}
  id={null}
  inputProps={null}
  inputSizeClass="col-md-4"
  label="Name"
  labelHelp={null}
  labelSizeClass="col-md-2"
  onChange={[Function]}
  required={true}
  type="text"
>
  <input
    className="form-control"
    id="common_parameter_name"
    name="common_parameter[name]"
    onChange={[Function]}
    required={true}
    type="text"
    value="something"
  />
</FormField>
`;
